src/org/zaproxy/zap/extension/pscanrulesAlpha/CSPNonceScanner.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2014 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Vector;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.Source;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+/**
+ * Invalid CSP Nonce passive scan rule
+ *
+ * @author
+ */
+public class CSPNonceScanner extends PluginPassiveScanner{
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.cspnonce.";
+	private static final int PLUGIN_ID = 19999; // TODO: get assigned an ID
+	private static final String CSP_HEADER = "Content-Security-Policy";
+
+	// https://w3c.github.io/webappsec-csp/#grammardef-nonce-source
+	private static final Pattern NONCE_SRC_PATT = Pattern.compile("nonce-([^ ,;]*)");

```

Comment:
This could produce false positives for domains containing `nonce-`. Should I rebase https://github.com/zaproxy/zap-extensions/pull/713 on to use that parser?